frontend/src/metabase/query_builder/components/notebook/NotebookStep/NotebookStep.unit.spec.tsx

```diff
@@ -76,4 +78,17 @@ describe("NotebookStep", () => {
       screen.queryByRole("button", { name: "Remove step" }),
     ).not.toBeInTheDocument();
   });
+
+  it("sets the row limit only on blur", () => {
+    const step = createMockNotebookStep({ type: "limit" });
+    const { updateQuery } = setup({ step });
+
+    const input = screen.getByPlaceholderText("Enter a limit");
+    userEvent.type(input, "38");
+    userEvent.type(input, "clear");
+    userEvent.type(input, "42");
+    input.blur();
```

Comment:
nit: same as above, I'd use `userEvent.tab()`, but we have plenty of places with input.blur, so it's up to you